modules/repository/license.go

```diff
@@ -111,3 +183,104 @@ func getLicensePlaceholder(name string) *licensePlaceholder {
 	}
 	return ret
 }
+
+// UpdateRepoLicensesByGitRepo will update repository licenses col if license file exists
+func UpdateRepoLicensesByGitRepo(ctx context.Context, repo *repo_model.Repository, gitRepo *git.Repository) error {
+	if gitRepo == nil {
+		var err error
+		gitRepo, err = git.OpenRepository(ctx, repo.RepoPath())
+		if err != nil {
+			return fmt.Errorf("OpenRepository: %w", err)
+		}
+		defer gitRepo.Close()
+	}
+	commit, err := gitRepo.GetBranchCommit(repo.DefaultBranch)
+	if err != nil {
+		if git.IsErrNotExist(err) {
+			// allow empty repo
+			return nil
+		}
+		return err
+	}
+	return UpdateRepoLicenses(ctx, repo, commit)
+}
+
+// UpdateRepoLicenses will update repository licenses col if license file exists
+func UpdateRepoLicenses(ctx context.Context, repo *repo_model.Repository, commit *git.Commit) error {
```

Comment:
How much time does it add to git push?

Why not add a queue that do check and update the repo in db async afterwards?

Reply:
This idea songs good! But I haven't touched queue before. I will take some times to try it.

Reply:
We have a good "framework" within gitea now, as we use it for e.g. automerge, pull-patch checking, ...

So you can just have a look at those

Reply:
Tried to use queue to handle the update process in the latest 3 commits.

Reply:
Thanks I'll try to review soon